### PR TITLE
CDAP-14450 docs for run count

### PR DESCRIPTION
**Pull request by yaojiefeng**
https://builds.cask.co/browse/CDAP-DQB432-1

### CODE REVIEW

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 1]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
        - | ``[{"runid":"cad83d45-ecfb-4bf8-8cdb-4928a5601b0e","start":1415051892,"end":1415057103,"status":"STOPPED"}]``
 
 
+Retrieving Run Records
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: Retrieving Run Records in Batch

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 2]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+----------------------
+
+To retrieve the latest run records for various applications and different program types, using an HTTP POST method::
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: run records for multiple programs, use::

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 3]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+
+  GET /v3/namespaces/<namespace-id>/runs
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: GET -> POST

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 4]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``namespace-id``
+     - Namespace ID
+
+with a JSON array in the request body consisting of multiple JSON objects with these parameters:
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: The request body must be a JSON array of objects with the following parameters:

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 5]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+   * - Parameter
+     - Description
+   * - ``appId``
+     - Name of the application being called
+   * - ``programType``
+     - One of ``mapreduce``, ``spark``, ``workflow``, ``flow``, ``service``, or ``worker``
+   * - ``programId``
+     - Name of the program (*mapreduce*, *spark*, *workflow*, *flow*, *service*, or *worker*) being called
+
+The response will be an array of JsonObjects each of which will contain the three input parameters
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: JSON Objects, each of which

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 6]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+as well as 2 fields, "runs" which is a list of the latest run records and "statusCode" which maps to the
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: as well as two of three possible extra fields -- "runs", which is a list of the latest run record for that program, "statusCode", which maps to the status code for retrieving the runs for that program, and "error" if there was an error retrieving runs for that program. The "statusCode" property will always be included, but "runs" and "error" are mutually exclusive.

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 7]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+status code for the data in that JsonObjects.
+If an error occurs in the input, then all JsonObjects for which the parameters have a valid status will
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: Remove this sentence

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 8]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``runs``
+     - Latest run records for the program defined by the individual JSON object's parameters
+   * - ``statusCode``
+     - The status code from retrieving the instance count of an individual JSON object
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: retrieving the program runs

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 9]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+message and statusCode.
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``runs``
+     - Latest run records for the program defined by the individual JSON object's parameters
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: The latest run records for the program

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 10]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+   * - ``statusCode``
+     - The status code from retrieving the instance count of an individual JSON object
+   * - ``error``
+     - If an error, a description of why the status was not retrieved (for example, the
+       specified program was not found, or the requested JSON object was missing a parameter)
+
+**Note:** The ``runs`` field is mutually exclusive of the ``error`` field.
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: The runs and error fields are mutually exclusive.

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 11]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+     - ``[{"appId": "App1", "programType": "Service", "programId": "Service1",``
+       ``"statusCode": 200, "runs": [...]},``
+       ``{"appId": "App1", "programType": "Workflow", "programId": "testWorkflow", "statusCode": 404,``
+       ``"error": "Program 'testWorkflow' is not found"},``
+       ``{"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow",``
+       ``"statusCode": 200, "runs": [...]]``
+   * - Description
+     - Attempt to retrieve the latest run records of the service *Service1* in the application *App1*,
+       the workflow *testWorkflow* in the application *App1*
+       and *MySvc1* in the application *MyApp3*, all in the namespace *default*
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: this does not match the example

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 12]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+
+
+Retrieving Run Counts
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: Retrieving Run Counts in Batch

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 13]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+---------------------
+
+To retrieve the run count for various applications and different program types, using an HTTP POST method::
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: To retrieve the run counts for multiple programs, use::

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 14]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+
+  GET /v3/namespaces/<namespace-id>/runcount
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: GET -> POST

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 15]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``namespace-id``
+     - Namespace ID
+
+with a JSON array in the request body consisting of multiple JSON objects with these parameters:
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: Make these parts consistent with the comments from above

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 16]
```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
...
+       the workflow *testWorkflow* in the application *App1*
+       and *MySvc1* in the application *MyApp3*, all in the namespace *default*
+
+
+Retrieving Specific Run Count
+-----------------------------
+
+To fetch the run count for a particular program, use::
+
+  GET /v3/namespaces/<namespace-id>/apps/{app-id}/{program-type}/{program-id}/runcount
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst:1762]: is it get or post? In the example, post is used.

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst (hunk 10 above)
Comment (yaojiefeng) [reply cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: I removed this line since it was already in the previous section

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**